src/waterlinked_dvl/waterlinked_dvl_tcp.cpp

```diff
@@ -303,6 +317,17 @@ void WaterlinkedDvlTcp::f_parse_json_v3(Json::Value root)
                 }
                 m_twist_publisher.publish(twist_msg);
             }
+            //Raw Twist
+            geometry_msgs::TwistWithCovarianceStamped twist_msg;
+            twist_msg.header = msg.header;
+            twist_msg.twist.twist.linear.x = msg.report.vx;
+            twist_msg.twist.twist.linear.y = msg.report.vy;
+            twist_msg.twist.twist.linear.z = msg.report.vz;
+
+            if (m_velocity_covariance.size() == twist_msg.twist.covariance.size()) {
```

Comment:
covariance source is wrong. it should rom the transducer report

Reply:
Fixed. Now mapping covariance values from transducer report to the v_dot submatrix in the covariance matrix.